src/tag/tag.scss

```diff
@@ -1,83 +1,88 @@
 // Copyright (c) Bentley Systems, Incorporated. All rights reserved.
 // See LICENSE.md in the project root for license terms and full copyright notice.
 @import '../style/index';
-@import '../icon/index';
 @import '../text/mixins';
+@import '../button/index';
 
-@mixin iui-tag {
+@mixin iui-tag-base {
   @include iui-reset;
   @include iui-focus($offset: -2px, $thickness: 2px);
   user-select: all;
   text-transform: lowercase;
   display: inline-flex;
+}
+
+@mixin iui-tag-basic {
+  @include iui-tag-base;
+  outline-offset: 1px;
 
   &:not(a) {
     @include themed {
       color: t(iui-text-color-muted);
     }
   }
+}
 
-  &:not(.iui-basic) {
-    height: $iui-baseline * 3;
-    margin: $iui-component-offset 0;
-    border-radius: ($iui-baseline * 3) * 0.5;
-    box-sizing: border-box;
-    padding: 0 $iui-xxs;
-    align-items: center;
-    @media (prefers-reduced-motion: no-preference) {
-      transition: border $iui-speed-fast ease-out;
-    }
-    font-size: $iui-font-size;
-    text-transform: none;
-    cursor: default;
-    @include themed {
-      border: 1px solid rgba(t(iui-color-foreground-body-rgb), t(iui-opacity-4));
-      background-color: t(iui-color-background-1);
-      color: t(iui-text-color);
-    }
-
-    > .iui-label {
-      @include iui-text;
-      white-space: nowrap;
-      text-overflow: ellipsis;
-      overflow: hidden;
-      margin: 0 $iui-xs 0 $iui-s;
+@mixin iui-tag {
+  @include iui-tag-base;
+  height: $iui-baseline * 3;
+  margin: $iui-component-offset 0;
+  border-radius: ($iui-baseline * 3) * 0.5;
+  box-sizing: border-box;
+  padding: 0 $iui-xxs;
+  align-items: center;
+  @media (prefers-reduced-motion: no-preference) {
+    transition: border $iui-speed-fast ease-out;
+  }
+  font-size: $iui-font-size;
+  text-transform: none;
+  cursor: default;
+  @include themed {
+    border: 1px solid rgba(t(iui-color-foreground-body-rgb), t(iui-opacity-4));
+    background-color: t(iui-color-background-1);
+    color: t(iui-text-color);
+  }
 
-      &:only-child {
-        margin: 0 $iui-s;
-      }
+  &:hover {
+    @include themed {
+      border-color: rgba(t(iui-color-foreground-body-rgb), t(iui-opacity-1));
     }
+  }
+}
 
-    > .iui-button {
-      border-radius: 50%;
```

Comment:
Hm, is it really worth it to have all button CSS when only border radius changed?

Reply:
I'll be honest, I'm going back and forth on this decision as well.  For this case, because the user can not pass any custom element in place of the close button, I vote we do this and not use the mixin:
```html
<span class="iui-tag">
	<span class="iui-tag-label">Hello</span>
	<button class="iui-button iui-borderless iui-small iui-tag-button">X</button>
</span>
```
```scss
.iui-tag-button {
	border-radius: $iui-border-radius-round;
}
```